### PR TITLE
[CBRD-24200] make LOG_WILL_COMMIT obsolete and delete corresponding codes

### PR DESCRIPTION
**Pull request by evelon**
http://jira.cubrid.org/browse/CBRD-24200

Purpose
make a macro obsolete that is not used anymore

Implementation
Make 'LOG_WILL_COMMIT' obsolete and delete all corresponding codes.

Remarks
N/A

### CODE REVIEW

File: src/transaction/log_record.hpp [hunk 1]
```diff
@@ -53,8 +53,8 @@ enum log_rectype
   LOG_CLIENT_USER_POSTPONE_DATA = 11,	/* Obsolete */
   LOG_RUN_NEXT_CLIENT_UNDO = 12,	/* Obsolete */
   LOG_RUN_NEXT_CLIENT_POSTPONE = 13,	/* Obsolete */
-#endif
   LOG_WILL_COMMIT = 14,		/* Transaction will be committed */
```
Comment (wrlawodms) [inline src/transaction/log_record.hpp]: better to edit comment like other obsoleted ones.
```suggestion
  LOG_WILL_COMMIT = 14,		/* Obsolete */
```

**Review by wrlawodms — COMMENTED** [review]
Looks good. Thank you for your work.

One suggestion: There are some changes because of removing trailing white space. This can confuse who wants to see the history later (with `git blame`). It would be better to get rid of unrelated changes. However, they are just comments so it's not that serious. Please fix it if it's not hard.

**Review by wrlawodms — APPROVED** [review]
Looks good to me!

**Review by hornetmj — APPROVED** [review]

**Review by joohok — APPROVED** [review]
LGTM.

**Review by eido5 — APPROVED** [review]
The verification is done. 
This function is used only in log_rv_analysis_will_commit().
Looks so good.

**Comment by hornetmj** [comment]
@mhoh3963 Please merge this PR. This is the unused enum value(LOG_WILL_COMMIT ) is obsoleted. So, the failures of test_sql and test_medium are not related to it.

**Pull request merged**